sympy/plotting/tests/test_series.py

```diff
@@ -0,0 +1,1635 @@
+    assert not np.isnan(x).any()
+    assert not np.isnan(y).any()
+
+
+def test_complex_range_line_plot_1():
+    # verify that univariate functions are evaluated with a complex
+    # data range (with zero imaginary part). There shouln't be any
+    # NaN value in the output.
+    if not np:
+        skip("numpy not installed.")
+
+    x, u = symbols("x, u")
+    expr1 = im(sqrt(x) * exp(-x**2))
+    expr2 = im(sqrt(u * x) * exp(-x**2))
+    s1 = LineOver1DRangeSeries(expr1, (x, -10, 10), adaptive=True,
+        adaptive_goal=0.1)
+    s2 = LineOver1DRangeSeries(expr1, (x, -10, 10), adaptive=False, n=30)
+    s3 = LineOver1DRangeSeries(expr2, (x, -10, 10), adaptive=False, n=30,
+        params={u: 1})
+
+    with warns(
+            RuntimeWarning,
+            match="invalid value encountered in sqrt",
+            test_stacklevel=False,
+        ):
+        data1 = s1.get_data()
+        data2 = s2.get_data()
+        data3 = s3.get_data()
+
+        assert not np.isnan(data1[1]).any()
+        assert not np.isnan(data2[1]).any()
+        assert not np.isnan(data3[1]).any()
+        assert np.allclose(data2[0], data3[0]) and np.allclose(data2[1], data3[1])
+
+
+@XFAIL
+def test_complex_range_line_plot_2():
+    # verify that univariate functions are evaluated with a complex
+    # data range (with non-zero imaginary part). There shouln't be any
+    # NaN value in the output.
+    if not np:
+        skip("numpy not installed.")
+
+    # NOTE: xfail because sympy's adaptive algorithm is unable to deal with
+    # complex number.
+
+    x, u = symbols("x, u")
+
+    # adaptive and uniform meshing should produce the same data.
+    # because of the adaptive nature, just compare the first and last points
+    # of both series.
+    s1 = LineOver1DRangeSeries(abs(sqrt(x)), (x, -5-2j, 5-2j), adaptive=True)
+    s2 = LineOver1DRangeSeries(abs(sqrt(x)), (x, -5-2j, 5-2j), adaptive=False,
+        n=10)
+    with warns(
+            RuntimeWarning,
+            match="invalid value encountered in sqrt",
+            test_stacklevel=False,
+        ):
+        d1 = s1.get_data()
+        d2 = s2.get_data()
+        xx1 = [d1[0][0], d1[0][-1]]
+        xx2 = [d2[0][0], d2[0][-1]]
+        yy1 = [d1[1][0], d1[1][-1]]
+        yy2 = [d2[1][0], d2[1][-1]]
+        assert np.allclose(xx1, xx2)
+        assert np.allclose(yy1, yy2)
+
+
+def test_force_real_eval():
+    # verify that force_real_eval=True produces inconsistent results when
+    # compared with evaluation of complex domain.
+    if not np:
+        skip("numpy not installed.")
+
+    x = symbols("x")
+
+    expr = im(sqrt(x) * exp(-x**2))
+    s1 = LineOver1DRangeSeries(expr, (x, -10, 10), adaptive=False, n=10,
+        force_real_eval=False)
+    s2 = LineOver1DRangeSeries(expr, (x, -10, 10), adaptive=False, n=10,
+        force_real_eval=True)
+    with warns(
+            RuntimeWarning,
+            match="invalid value encountered in sqrt",
+            test_stacklevel=False,
+        ):
```

Comment:
@oscarbenjamin These lines of code catches a warning just fine with pytest. However, tests are failing with pyodide: apparently, no warning is raised there... Is there any way to code  "maybe this raise a warning"?

Reply:
The `warns` context manager is used to assert that a warning is emitted.

We also have the `ignore_warnings` context manager which suppresses warnings but does not assert that any warnings are emitted.

Reply:
Thank you! :)